Tools/clinic/clinic.py

```diff
@@ -4743,23 +4744,21 @@ def state_modulename_name(self, line: str | None) -> None:
         return_converter = None
         if returns:
             ast_input = f"def x() -> {returns}: pass"
-            module = None
             try:
-                module = ast.parse(ast_input)
+                module_node = ast.parse(ast_input)
             except SyntaxError:
-                pass
-            if not module:
-                fail("Badly-formed annotation for " + full_name + ": " + returns)
+                fail(f"Badly formed annotation for {full_name}: {returns!r}")
+            function_node = module_node.body[0]
+            assert isinstance(function_node, ast.FunctionDef)
             try:
-                name, legacy, kwargs = self.parse_converter(module.body[0].returns)
+                name, legacy, kwargs = self.parse_converter(function_node.returns)
                 if legacy:
-                    fail("Legacy converter {!r} not allowed as a return converter"
-                         .format(name))
+                    fail(f"Legacy converter {name!r} not allowed as a return converter")
                 if name not in return_converters:
-                    fail("No available return converter called " + repr(name))
+                    fail(f"No available return converter called {name!r}")
                 return_converter = return_converters[name](**kwargs)
             except ValueError:
-                fail("Badly-formed annotation for " + full_name + ": " + returns)
+                fail(f"Badly formed annotation for {full_name}: {returns!r}")
```

Comment:
I've added tests for the other three failure paths that I'm touching in this PR, but I can't figure out how to trigger the `ValueError` here -- any ideas @erlend-aasland?

Reply:
Nope; I also tried, but failed to trigger it :(

Reply:
Ugh, for me, that puts the `try/except ValueError` in the unfortunate place of "code I don't know how to add tests for but am also too scared to remove in case removing it breaks something" ://